web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -27,23 +27,47 @@ export async function sendAndConfirmTransaction(
     maxRetries: options.maxRetries,
   };
 
+  let status;
+
   const signature = await connection.sendTransaction(
     transaction,
     signers,
     sendOptions,
   );
 
-  const status = (
+  status =
+    transaction.recentBlockhash != null &&
+    transaction.lastValidBlockHeight != null
+      ? (
+          await connection.confirmTransaction(
+            {
+              signature: signature,
+              blockhash: transaction.recentBlockhash!,
+              lastValidBlockHeight: transaction.lastValidBlockHeight!,
+            },
+            options && options.commitment,
+          )
+        ).value
+      : (
+          await connection.confirmTransaction(
+            signature,
+            options && options.commitment,
+          )
+        ).value;
+
+  status = (
     await connection.confirmTransaction(
-      signature,
+      {
+        signature: signature,
+        blockhash: transaction.recentBlockhash!,
+        lastValidBlockHeight: transaction.lastValidBlockHeight!,
+      },
       options && options.commitment,
     )
   ).value;
```

Comment:
Oh oops, I think this one got accidentally left in here?

Reply:
Yep, I did indeed forget to delete that. Got it now.